WooCommerce/Classes/POS/Utils/POSProductProvider.swift

```diff
@@ -0,0 +1,90 @@
+import Foundation
+import Yosemite
+import protocol Storage.StorageManagerType
+import class WooFoundation.CurrencySettings
+
+/// Product provider for the Point of Sale feature
+///
+final class POSProductProvider {
```

Comment:
💭 non-blocking thought: how about having this provider (and `POSProduct` or equivalent) in Yosemite (can create a POS folder there as if it were a "POS data layer framework"), potentially in a protocol with `func providePointOfSaleProducts() -> [POSProduct]` and maybe other preview helpers, where the POS folder only knows about the protocol? Then the product provider implementation is DI'ed to `PointOfSaleEntryPointView` to the view model. This would separate the POS implementation details from the `app`/its POS folder, like the usage of Core Data through `ResultsController` (which is already in Yosemite).

Reply:
That sounds pretty good, and follows the idea of having a root of composition for all these dependencies, keeping things separate, and disallowing the view of creating its own dependencies. I'll work on this separately 🙇